src/bin/pgcopydb/ld_transform.c

```diff
@@ -1169,6 +1169,64 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 }
 
 
+/* append the new insert statement if targets same table as previous */
+static bool
+coalesceInsertStatements(LogicalMessageInsert *last,
+						 LogicalMessageInsert *stmt)
+{
+	if (strcmp(last->nspname, stmt->nspname) != 0 ||
+		strcmp(last->relname, stmt->relname) != 0)
+	{
+		return false;
+	}
+
+	LogicalMessageValuesArray *lastValuesArray = &last->new.array->values;
+
+	if (lastValuesArray->capacity < (lastValuesArray->count + 1))
+	{
+		/* double the capacity to avoid excessive realloc */
+		lastValuesArray->capacity *= 2;
```

Comment:
We should also check that we are not pushing the number of EXECUTE arguments above the limit. I seem to remember that you can have as many as 16384 arguments (int4).

Reply:
Looks like type of `nParams` is `int`. Do you mean the limit is imposed in the server side?
```
PGresult *PQexecPrepared(PGconn *conn,
                         const char *stmtName,
                         int nParams,
                         const char * const *paramValues,
                         const int *paramLengths,
                         const int *paramFormats,
                         int resultFormat);
```

Reply:
Have a look at the protocol definition for the Parse message at https://www.postgresql.org/docs/16/protocol-message-formats.html: 

> **Int16** The number of parameter data types specified (can be zero). Note that this is not an indication of the number of parameters that might appear in the query string, only the number that the frontend wants to prespecify types for.

The number of parameters must fit in an Int16 in the protocol, hence the limit I was sharing.